source/DSCResources/DSC_CMForestDiscovery/DSC_CMForestDiscovery.psm1

```diff
@@ -0,0 +1,299 @@
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $PollingSchedule,
+
+        [Parameter()]
+        [Boolean]
+        $EnableActiveDirectorySiteBoundaryCreation,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSubnetBoundaryCreation
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+        if ($Enabled -eq $true)
+        {
+            $exludelist = @('Verbose','PollingSchedule')
+            foreach ($param in $PSBoundParameters.GetEnumerator())
+            {
+                if ($exludelist -notcontains $param.key)
+                {
+                    if ($param.Value -ne $state[$param.key])
+                    {
+                        Write-Verbose -Message ($script:localizedData.SettingSettings -f $param.Key, $param.Value)
+                        $buildingParams += @{
+                            $param.Key = $param.Value
+                        }
+                    }
+                }
+            }
+
+            if (-not [string]::IsNullOrEmpty($PollingSchedule))
+            {
+                $newSchedule = @{
+                    RecurInterval = $PollingSchedule.RecurInterval
+                    RecurCount    = $PollingSchedule.RecurCount
+                }
+
+                $desiredPollingSchedule = New-CMSchedule @newSchedule
+
+                $currentSchedule = @{
+                    RecurInterval = $state.PollingSchedule.RecurInterval
+                    RecurCount    = $state.PollingSchedule.RecurCount
+                }
+
+                $stateSchedule = New-CMSchedule @currentSchedule
+
+                $array = @('DayDuration','DaySpan','HourDuration','HourSpan','IsGMT','MinuteDuration','MinuteSpan')
+
+                foreach ($item in $array)
+                {
+                    if (($desiredPollingSchedule).$($item) -ne ($stateSchedule).$($item))
+                    {
+                        Write-Verbose -Message ($script:localizedData.ScheduleItem `
+                            -f $item, $($desiredPollingSchedule.$($item)), $(($stateSchedule).$($item)))
+                        $setSchedule = $true
+                    }
+                }
+
+                if ($setSchedule)
+                {
+                    $buildingParams += @{
+                        PollingSchedule = $desiredPollingSchedule
+                    }
+                }
+            }
+
+            if ($buildingParams)
+            {
+                Set-CMDiscoveryMethod -ActiveDirectoryForestDiscovery -SiteCode $SiteCode @buildingParams
+            }
+        }
+        else
```

Comment:
This could be changed to an elseif ($Enabled -eq $true -and $state.Enabled -eq $true)

Reply:
updated, except we want it to be elseif ($Enabled -eq $false -and $state.Enabled -eq $true)